python/packages/autogen-core/src/autogen_core/components/tools/_function_tool.py

```diff
@@ -13,7 +13,58 @@
 
 
 class FunctionTool(BaseTool[BaseModel, BaseModel]):
+    """
+    Create custom tools by wrapping standard Python functions.
+
+    `FunctionTool` offers an interface for executing Python functions either asynchronously or synchronously.
+    Each function must include type annotations for all parameters and its return type. These annotations
+    enable `FunctionTool` to generate a schema necessary for input validation, serialization, and for informing
+    the LLM about expected parameters. When the LLM prepares a function call, it leverages this schema to
+    generate arguments that align with the function's specifications.
+
+    .. note::
+
+        It is the user's responsibility to verify that the tool's output type matches the expected type.
+
+    Example:
+
+        .. code-block:: python
+
+            import random
+            from autogen_core.base import CancellationToken
+            from autogen_core.components.tools import FunctionTool
+            from typing_extensions import Annotated
+
+
+            async def get_stock_price(ticker: str, date: Annotated[str, "Date in YYYY/MM/DD"]) -> float:
+                # Simulates a stock price retrieval by returning a random float within a specified range.
+                return random.uniform(10, 200)
+
+
+            # Initialize a FunctionTool instance for retrieving stock prices.
+            stock_price_tool = FunctionTool(get_stock_price, description="Fetch the stock price for a given ticker.")
+
+            # Execute the tool with cancellation support.
+            cancellation_token = CancellationToken()
+            result = await stock_price_tool.run_json({"ticker": "AAPL", "date": "2021/01/01"}, cancellation_token)
+
+            # Output the result as a formatted string.
+            print(stock_price_tool.return_value_as_string(result))
+    """
+
     def __init__(self, func: Callable[..., Any], description: str, name: str | None = None) -> None:
+        """
```

Comment:
Let's use the class docstring for the `__init__` doc string. i.e., moving the `Args` docs to the class docstring.